model-optimizer/extensions/front/broadcast_with_range.py

```diff
@@ -51,16 +53,26 @@ def replace(node: Node, const: Node):
 
         positive_idx = non_one_dims.item(0)
         negative_idx = positive_idx - len(shape)
+
+        node_name = node.soft_get('name', node.id)
         gather = create_op_with_const_inputs(graph, Gather, {1: int64_array(negative_idx), 2: int64_array(0)},
-                                             {'name': node.soft_get('name', node.id) + '/BroadcastingDim'})
+                                             {'name': node_name + '/BroadcastingDim'})
 
         range_node = create_op_with_const_inputs(graph, Range,
                                                  {0: np.array(0, dtype=value.dtype),
                                                   2: np.array(1, dtype=value.dtype)},
                                                  {'name': const_name + '/Range', 'dtype': value.dtype})
 
-        node.in_port(1).get_connection().add_destination(gather.in_port(0))
+        shapeof0_node = Shape(graph, {'name': const_name + '/ShapeOf'}).create_node()
+        shapeof1_node = Shape(graph, {'name': node_name + '/ShapeOf'}).create_node()
+        max_node = Maximum(graph, {'name': node_name + '/MaxInputShape'}).create_node([shapeof0_node, shapeof1_node])
```

Comment:
what would happen if shapes would be represented as tensors of shape [3] and [4]? Maximum shape inference would fail

Reply:
Done